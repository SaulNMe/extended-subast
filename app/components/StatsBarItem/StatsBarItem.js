import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {
	Text,
	View
} from 'react-native';

import style from './StatsBarItemStyle';
import SubtitleText from 'cotizame-native/app/components/SubtitleText';
import SmallText from 'cotizame-native/app/components/SmallText';

export default class StatsBarItem extends Component {
	render = () => (
		<View style={style.container}>
			<SubtitleText
				text= {this.props.data}
				color='white'
				weight='regular'
			/>
			<SmallText
				text= {this.props.title}
				color='white'
				weight='regular'
			/>
		</View>
	)
}

StatsBarItem.propTypes = {
	data: PropTypes.string,
	title: PropTypes.string
}

StatsBarItem.defaultProps = {
	data: "- -",
	title: "- -"
}
