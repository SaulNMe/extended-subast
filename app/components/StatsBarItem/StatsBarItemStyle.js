import { StyleSheet } from 'react-native';
import { Fonts, Colors, Metrics } from 'cotizame-native/app/styles';
import { ApplicationStyles } from 'cotizame-native/app/styles';

export default StyleSheet.create({
	container:{
		justifyContent: "center",
		alignItems: "center",
		margin: 12
	}
});
