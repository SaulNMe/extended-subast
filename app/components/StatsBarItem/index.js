import StatsBarItem from  './StatsBarItem';

export default StatsBarItem;
