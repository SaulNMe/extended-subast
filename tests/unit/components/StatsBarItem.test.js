import React from 'react';
import { shallow } from 'enzyme';

import StatsBarItem from 'cotizame-native/app/components/StatsBarItem';

function createTestProps (props) {
	return {
		// common props
		data: '',
		title: '',
		// allow to override common props
		...props,
	}
}

describe('redering', ()=> {
	let wrapper;
	const createWrapper = props => shallow(<StatsBarItem {...props} />);
	beforeEach(()=>{
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', ()=>{
		expect(wrapper).toMatchSnapshot();
	});
	describe('whit default props', ()=>{
		beforeEach(()=>{
			wrapper = createWrapper()
		})
		it('data is correct', () => {
			let props = wrapper.instance().props;
			expect(props.data).toBe('- -');
		});
		it('title is correct', () => {
			let props = wrapper.instance().props;
			expect(props.title).toBe('- -');

		})
	})


})