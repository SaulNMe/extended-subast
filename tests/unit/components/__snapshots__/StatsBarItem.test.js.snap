// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redering renders as expected 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "margin": 12,
    }
  }
>
  <SubtitleText
    color="darker"
    text=""
    weight="regular"
  />
  <SmallText
    color="darker"
    text=""
    weight="regular"
  />
</Component>
`;
